internal/expr/parser.go

```diff
@@ -392,47 +426,36 @@ func (p *Parser) parseColumn() (columnExpr, bool, error) {
 	return fullName{}, false, nil
 }
 
-func (p *Parser) parseFuncName() (columnExpr, bool, error) {
+func (p *Parser) parseFunc() (columnExpr, bool, error) {
 	cp := p.save()
-	if p.skipName() && p.skipByte('(') {
-		bracketLevel := 1
+	scp := p.startSubpart()
+	if p.skipName() && p.peekByte('(') {
```

Comment:
The change to `peekByte` here feels slightly off, is this because `advance` is needed to handle the change in bracket level? Perhaps `skipByte` should handle this too

Reply:
So theres actually another reason this is here. If an input expression starts right after the bracket then advance won't notice its there. It checks if the preceding character is of interest so if it starts on a '$' then it won't stop since it would be expecting to parse an input expression on the next character, i.e. the one after the '$'.

I agree its not amazing to have `peekByte` there, but the other solutions that don't involve re-implementing the main parser are all more ugly.